web-registry/src/apollo.tsx

```diff
@@ -1,84 +1,60 @@
-import { useCallback, useEffect } from 'react';
-import {
-  ApolloClient,
-  ApolloLink,
-  ApolloProvider,
-  HttpLink,
-  InMemoryCache,
-} from '@apollo/client';
+import { ApolloProvider, createHttpLink, InMemoryCache } from '@apollo/client';
 import { setContext } from '@apollo/client/link/context';
-import { useAuth0 } from '@auth0/auth0-react';
+import { useQuery } from '@tanstack/react-query';
 
-import getApiUri from './lib/apiUri';
+import getApiUri from 'lib/apiUri';
+import { ApolloClientFactory } from 'lib/clients/apolloClientFactory';
 
 interface AuthApolloProviderProps {
+  apolloClientFactory: ApolloClientFactory;
   children?: any;
 }
 
 export const AuthApolloProvider = ({
+  apolloClientFactory,
   children,
 }: AuthApolloProviderProps): any => {
-  const { user, isLoading, isAuthenticated, getAccessTokenSilently } =
-    useAuth0();
-  const apiUri = getApiUri();
+  const baseUri = getApiUri();
+  const query = useQuery({
+    queryKey: ['csrfToken'],
+    queryFn: async () => {
+      const resp = await fetch(`${baseUri}/csrfToken`, {
+        method: 'GET',
+        credentials: 'include',
+      });
+      const { token } = await resp.json();
+      return { token, headers: resp.headers };
+    },
+  });
```

Comment:
maybe we could extract this part into a reusable query hook?
anyway I can do that as part of https://github.com/regen-network/regen-web/pull/1814 too so we can merge this

Reply:
SGTM, let's do this there

Reply:
done: https://github.com/regen-network/regen-web/pull/1814/commits/59e4f853776fbba9ab25d4c2a35e55cc2cf3782b